src/main/java/org/folio/tag/repository/TagRepositoryImpl.java

```diff
@@ -0,0 +1,245 @@
+package org.folio.tag.repository;
+
+import static java.util.Arrays.asList;
+
+import static org.folio.tag.repository.TagTableConstants.DELETE_TAG_RECORD;
+import static org.folio.tag.repository.TagTableConstants.ID_COLUMN;
+import static org.folio.tag.repository.TagTableConstants.RECORD_ID_COLUMN;
+import static org.folio.tag.repository.TagTableConstants.RECORD_TYPE_COLUMN;
+import static org.folio.tag.repository.TagTableConstants.SELECT_ALL_TAGS;
+import static org.folio.tag.repository.TagTableConstants.SELECT_TAGS_BY_RECORD_ID_AND_RECORD_TYPE;
+import static org.folio.tag.repository.TagTableConstants.SELECT_TAGS_BY_RECORD_TYPES;
+import static org.folio.tag.repository.TagTableConstants.TABLE_NAME;
+import static org.folio.tag.repository.TagTableConstants.TAG_COLUMN;
+import static org.folio.tag.repository.TagTableConstants.UPDATE_INSERT_STATEMENT_FOR_PROVIDER;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.ResultSet;
+import io.vertx.ext.sql.SQLConnection;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import org.folio.common.FutureUtils;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.tag.RecordType;
+import org.folio.tag.Tag;
+
+@Component
+class TagRepositoryImpl implements TagRepository {
+  
+  private static final Logger LOG = LoggerFactory.getLogger(TagRepositoryImpl.class);
+  private Vertx vertx;
+
+  @Autowired
+  public TagRepositoryImpl(Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  @Override
+  public CompletableFuture<List<Tag>> findAll(String tenantId) {
+    Future<ResultSet> resultSetFuture = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .select(
+        String.format(SELECT_ALL_TAGS, getTableName(tenantId)),
+        resultSetFuture.completer()
+      );
+
+    return mapResultSet(resultSetFuture, this::readTags);
+  }
+
+  @Override
+  public CompletableFuture<List<Tag>> findByRecord(String tenantId, String recordId, RecordType recordType) {
+    JsonArray parameters = createParameters(asList(recordId, recordType.getValue()));
+
+    Future<ResultSet> resultSetFuture = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .select(
+        String.format(SELECT_TAGS_BY_RECORD_ID_AND_RECORD_TYPE, getTableName(tenantId)),
+        parameters, resultSetFuture.completer()
```

Comment:
complete?

Reply:
it's a handler result of the future:
![image](https://user-images.githubusercontent.com/42244720/54816865-1b6a3600-4c9e-11e9-9110-284e69936c33.png)
can be simply a future, but to emphasize it's actually a handler I used `completer()` method

Reply:
👍